engine/src/Format.inl

```diff
@@ -5,7 +5,7 @@ std::string Engine::Core::Tools::stringFormat(const std::string &format, Args...
 {
     int size = snprintf(nullptr, 0, format.c_str(), args...) + 1; // Extra space for '\0'
 
-    assert(size > 0, "Error during formatting.");
+    assert(size > 0 || "Error during formatting.");
```

Comment:
Why ? What is the purpose of the operator || ? 
While trying to understand, I tested it and found out the assert never asserts with that.
```cpp
assert(false || "comment");
``` 
This code never asserts for example.
It's because the pointer to the string "comment" is not null, so it is "true", and so, the final result is also true.
Please fix.

Reply:
that true, it's an && instand